### PR TITLE
Enhanced RDoc for Enumerable

### PR DESCRIPTION
**Pull request by BurdetteLamar**
Treats:
- #min
- #max
- #minmax
- #min_by
- #max_by
- #minmax_by
- #include?

### CODE REVIEW

File: enum.c [hunk 1]
```diff
@@ -2624,17 +2648,25 @@ minmax_by_i(RB_BLOCK_CALL_FUNC_ARGLIST(i, _memo))
...
- *     enum.minmax_by { |obj| block } -> [min, max]
- *     enum.minmax_by                 -> an_enumerator
+ *    minmax_by {|element| ... } -> [minimum, maximum]
+ *    minmax_by                  -> enumerator
  *
- *  Returns a two element array containing the objects in
- *  <i>enum</i> that correspond to the minimum and maximum values respectively
- *  from the given block.
+ *  Returns a 2-element array containing the elements
+ *  for which the block returns minimum an maximum values:
```
Comment (jeremyevans) [inline enum.c:2655]: `minimum an maximum` -> `minimum and maximum`

**Review by jeremyevans — CHANGES_REQUESTED** [review]
Looks good. One small typo fix, then it can be committed without further review.

File: enum.c (hunk 1 above)
Comment (BurdetteLamar) [reply enum.c]: Done.

File: enum.c [hunk 2]
```diff
@@ -2181,26 +2183,54 @@ max_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
+ *
+ *  - A negative integer if <tt>a < b</tt>.
+ *  - Zero if <tt>a == b</tt>.
+ *  - A positive integer if <tt>a > b</tt>.
+ *
+ *  With a block given and no argument,
+ *  returns the maximum element as determined by the block:
+ *
+ *    %w[xxx x xxxx xx].max {|a, b| a.size <=> b.size } # => "xxxx"
+ *    [].max {|a, b| a <=> b }                          # => nil
```
Comment (JuPlutonic) [inline enum.c:2223]: suggestion:
`{foo: 0, bar: 1, baz: 2}.max {|pair1, pair2| pair1[1] <=> pair2[1] } # => [:baz, 2]`

File: enum.c [hunk 3]
```diff
@@ -2181,26 +2183,54 @@ max_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
+ *  The block is called with two elements +a+ and +b+, and must return:
+ *
+ *  - A negative integer if <tt>a < b</tt>.
+ *  - Zero if <tt>a == b</tt>.
+ *  - A positive integer if <tt>a > b</tt>.
+ *
+ *  With a block given and no argument,
+ *  returns the maximum element as determined by the block:
+ *
+ *    %w[xxx x xxxx xx].max {|a, b| a.size <=> b.size } # => "xxxx"
```
Comment (JuPlutonic) [inline enum.c:2222]: suggestion:
`{foo: 0, bar: 1, baz: 2}.max {|pair1, pair2| pair1[1] <=> pair2[1] } # => [:baz, 2]`

File: enum.c [hunk 4]
```diff
@@ -2181,26 +2183,54 @@ max_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, args))
...
+ *    [].max {|a, b| a <=> b }                          # => nil
+ *
+ *  With a block given and positive integer argument +n+ given,
+ *  returns an array containing the first +n+ maximum elements that exist,
+ *  as determined by the block.
+ *
+ *    %w[xxx x xxxx xx].max(2) {|a, b| a.size <=> b.size } # => ["xxxx", "xxx"]
```
Comment (JuPlutonic) [inline enum.c:2229]: suggestion:
`{foo: 0, bar: 1, baz: 2}.max(2) {|pair1, pair2| pair1[1] <=> pair2[1] } # => [[:baz, 2], [:bar, 1]]`

File: enum.c [hunk 5]
```diff
@@ -2341,17 +2371,30 @@ minmax_ii(RB_BLOCK_CALL_FUNC_ARGLIST(i, _memo))
...
+ *    (1..4).minmax                   # => [1, 4]
+ *    (-4..-1).minmax                 # => [-4, -1]
+ *    %w[d c b a].minmax              # => ["a", "d"]
+ *    {foo: 0, bar: 1, baz: 2}.minmax # => [[:bar, 1], [:foo, 0]]
+ *    [].minmax                       # => [nil, nil]
+ *
+ *  With a block given, returns the minimum and maximum elements
+ *  as determined by the block:
+ *
+ *    %w[xxx x xxxx xx].minmax {|a, b| a.size <=> b.size } # => ["x", "xxxx"]
```
Comment (JuPlutonic) [inline enum.c:2393]: suggestion:
`{foo: 0, bar: 1, baz: 2}.minmax {|pair1, pair2| pair1[1] <=> pair2[1] } # => [[:foo, 0], [:baz, 2]]`

**Comment by BurdetteLamar** [comment]
Thanks, @JuPlutonic, I will look at these comments.

**Pull request merged**